src/compiler/compile/render_dom/wrappers/shared/is_head.ts

```diff
@@ -1,3 +1,9 @@
-export function is_head(node) {
-	return node && node.type === 'MemberExpression' && node.object.name === '@_document' && node.property.name === 'head';
+import { Identifier, Node } from 'estree';
+
+type NarrowNodeType<Y> = Y extends { name: string } ? Y : never;
+
+export function is_head(node: Identifier | Node) {
+	return node && node.type === 'MemberExpression' &&
+		(node.object as NarrowNodeType<Node>).name === '@_document' &&
+		(node.property as NarrowNodeType<Node>).name === 'head';
```

Comment:
[imo]

Node that has `name` are only `Identifier` and `PrivateIdentifier`.
So if `name` does not exist, `node.object.name` returns undefined.
I think here checks that Node type is `Identifier` or a `PrivateIdentifier` AND `name` has the expected value.

Therefore, I thought a more meaningful fix here would be something like `node.object['name']`.

Reply:
Yeah, I was not sure how to improve this since I don't know the `estree` types that well.
Your solution makes much more sense. Thanks!